File: Striver/Trees/Q04.java
package Striver.Trees;

import java.util.*;

public class Q04 {
    // Level Order Traversal
    public static List<List<Integer>> levelOrder(Node root){
        List<List<Integer>> ans = new ArrayList<>();
        if(root == null) return ans;

        Queue<Node> q = new LinkedList<>();
        q.add(root);

        while(!q.isEmpty()){
            int size = q.size();
            List<Integer> level = new ArrayList<>();
            for(int i = 0; i < size; i++){
                Node curr = q.poll();
                level.add(curr.data);
                if(curr.left != null) q.add(curr.left);
                if(curr.right != null) q.add(curr.right);
            }
            ans.add(level);
        }
        return ans;
    }

    public static void main(String[] args) {
        Node root = new Node(1);
        root.left = new Node(2);
        root.right = new Node(3);
        root.left.left = new Node(4);
        root.left.right = new Node(5);
        root.right.left = new Node(6);
        root.right.right = new Node(7);

        List<List<Integer>> ans = levelOrder(root);
        System.out.println(ans);
    }
}

File: Striver/Trees/Q04.py
# level order traversal of a tree
from Node import Node

def levelOrder(root):
    ans = []

    if root is None:
        return ans
    
    queue = []
    queue.append(root)

    while len(queue) > 0:
        level = []
        for i in range(len(queue)):
            curr = queue.pop(0)
            level.append(curr.data)
            if curr.left is not None:
                queue.append(curr.left)
            if curr.right is not None:
                queue.append(curr.right)
        ans.append(level)
    
    return ans

root = Node(3)
root.left = Node(9)
root.right = Node(20)
root.right.left = Node(15)
root.right.right = Node(7)

print(levelOrder(root))